libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -549,679 +452,290 @@ - (void)migrateUserDefaults {
-                    } else {
-                        // Error logging will already have occurred.  Make sure account file data is removed.
-                        [fm removeItemAtPath:userAccountPath error:nil];
-                    }
-                } else {
-                    [self log:SFLogLevelDebug format:@"There is no user account file in this user directory: %@", orgPath];
-                }
-            }
-        }
-    }
-    
-    // Convert any legacy active user data to the active user identity.
-    [SFUserAccountManagerUpgrade updateToActiveUserIdentity:self];
-    
-    SFUserAccountIdentity *curUserIdentity = self.activeUserIdentity;
-    
-    // In case the most recently used account was removed, or the most recent account is the temporary account,
-    // see if we can load another available account.
-    if (nil == curUserIdentity || [curUserIdentity isEqual:self.temporaryUserIdentity]) {
-        for (SFUserAccount *account in self.userAccountMap.allValues) {
-            if (account.credentials.userId) {
-                curUserIdentity = account.accountIdentity;
-                break;
-            }
-        }
-    }
-    if (nil == curUserIdentity) {
-        [self log:SFLogLevelInfo msg:@"Current active user identity is nil"];
-    }
-    
-    self.previousCommunityId = self.activeCommunityId;
 
-    SFUserAccount *account = nil;
-    
-    account = [self desiredAccountWithPreviousIdentity:curUserIdentity];
-    
-    if (account) {
-        curUserIdentity = account.accountIdentity;
-        self.previousCommunityId = nil;
-    }
-    
-    if (curUserIdentity){
-        account = [self userAccountForUserIdentity:curUserIdentity];
-        account.communityId = self.previousCommunityId;
-        self.currentUser = account;
-    } else {
-        self.currentUser = nil;
-    }
-    
-    // update the client ID in case it's changed (via settings, etc)
-    self.currentUser.credentials.clientId = self.oauthClientId;
-    
-    [self userChanged:SFUserAccountChangeCredentials];
-    
-    return YES;
 }
 
-- (SFUserAccount *)loadUserAccountFromFile:(NSString *)filePath {
-    NSFileManager *manager = [[NSFileManager alloc] init];
-    if (![manager fileExistsAtPath:filePath]) {
-        [self log:SFLogLevelDebug format:@"No account data exists at '%@'", filePath];
-        return nil;
-    }
+- (BOOL)loadAccounts:(NSError **) error {
+    BOOL success = YES;
+    [accountsLock lock];
 
-    // Try to load the user account assuming it is encrypted
-    SFUserAccount *user = nil;
-    if ([self loadUserAccountFromFile:filePath encrypted:YES account:&user error:nil]) {
-        return user;
-    } else {
+    NSError *internalError = nil;
+    NSDictionary<SFUserAccountIdentity *,SFUserAccount *> *accounts = [accountPersister fetchAllAccounts:&internalError];
+    [_userAccountMap removeAllObjects];
+    _userAccountMap = [NSMutableDictionary  dictionaryWithDictionary:accounts];
```

Comment:
I guess `fetchAllAccounts:` commits to returning a `nonnull` value, so it's a moot point.  Never mind!